tools/build-samples-and-snippets.ps1

```diff
@@ -93,13 +93,27 @@ foreach($sample in $samples) {
     Set-Location -Path $sample.Directory.FullName
     Get-ChildItem -inc bin,obj -rec | Remove-Item -rec -force
     
-    msbuild $sample.Name -nodeReuse:true -verbosity:minimal -restore -property:RestorePackagesConfig=true
-	
-	if( -not $? ) {
-		$exitCode = 1
-		Write-Output ("::error::Build failed: {0}" -f $sample.FullName)
-		$failedProjects.Add($sample.FullName)
-	}
+    try 
+    {
+        if(Test-Path msbuild)
+        {
+            msbuild $sample.Name -nodeReuse:true -verbosity:minimal -restore -property:RestorePackagesConfig=true
+        }
+        else 
+        {
+            dotnet build $sample.Name -nodeReuse:true -verbosity:minimal -restore -property:RestorePackagesConfig=true
+        }
+        
+        if( -not $? ) {
+            $exitCode = 1
+            Write-Output ("::error::Build failed: {0}" -f $sample.FullName)
+            $failedProjects.Add($sample.FullName)
+        }
+    } finally 
```

Comment:
Makes also sure when you hit Ctrl+C locally during build it restores the root folder of the docs directory as the current directory